src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -205,6 +206,10 @@ public function handleChargeRefunded(Event $event): Response
             return $this->validationError(sprintf('Unsupported Status "%s"', $event->data->object->status));
         }
 
+        if ($donation->isReversed() && $event->data->object->metadata->matchedAmount > 0) {
```

Comment:
It feels safer, and more resilient to the fact hooks could come in in any order, to base this decision on our current local status rather than what's in the hook.

`if ($donation->isReversed() && $donation->getCampaign()->isMatched()) { ...` is used elsewhere and should be safe even if the actual matched amount is £0 I think. We could also do a check of the amount from the local txn info.

Reply:
@NoelLH good shout, I've updated this with the suggested stricter checks before releasing match funds where the amount refunded must match the local amount. For partial refunds (tip refunds) I guess it'll belong to the same logic branch but we don't release match funds, probably one for another time once we have BDD written for this.